### PR TITLE
ffmpeg_2: mark as insecure

### LINKED ISSUE #93984: Vulnerability roundup 90: ffmpeg-2.8.17: 41 advisories [9.8]
[search](https://search.nix.gsc.io/?q=ffmpeg&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=ffmpeg+in%3Apath&type=Code)

* [ ] [CVE-2017-7859](https://nvd.nist.gov/vuln/detail/CVE-2017-7859) CVSSv3=9.8 (nixos-unstable)
* [ ] [CVE-2018-1999010](https://nvd.nist.gov/vuln/detail/CVE-2018-1999010) CVSSv3=9.8 (nixos-unstable)
* [ ] [CVE-2019-12730](https://nvd.nist.gov/vuln/detail/CVE-2019-12730) CVSSv3=9.8 (nixos-unstable)
* [ ] [CVE-2019-17539](https://nvd.nist.gov/vuln/detail/CVE-2019-17539) CVSSv3=9.8 (nixos-unstable)
* [ ] [CVE-2019-17542](https://nvd.nist.gov/vuln/detail/CVE-2019-17542) CVSSv3=9.8 (nixos-unstable)
* [ ] [CVE-2017-9990](https://nvd.nist.gov/vuln/detail/CVE-2017-9990) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2017-14767](https://nvd.nist.gov/vuln/detail/CVE-2017-14767) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2017-15672](https://nvd.nist.gov/vuln/detail/CVE-2017-15672) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2018-9841](https://nvd.nist.gov/vuln/detail/CVE-2018-9841) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2018-1999011](https://nvd.nist.gov/vuln/detail/CVE-2018-1999011) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2019-15942](https://nvd.nist.gov/vuln/detail/CVE-2019-15942) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2020-14212](https://nvd.nist.gov/vuln/detail/CVE-2020-14212) CVSSv3=8.8 (nixos-unstable)
* [ ] [CVE-2016-6671](https://nvd.nist.gov/vuln/detail/CVE-2016-6671) CVSSv3=7.8 (nixos-unstable)
* [ ] [CVE-2016-7450](https://nvd.nist.gov/vuln/detail/CVE-2016-7450) CVSSv3=7.8 (nixos-unstable)
* [ ] [CVE-2016-7502](https://nvd.nist.gov/vuln/detail/CVE-2016-7502) CVSSv3=7.8 (nixos-unstable)
* [ ] [CVE-2017-11399](https://nvd.nist.gov/vuln/detail/CVE-2017-11399) CVSSv3=7.8 (nixos-unstable)
* [ ] [CVE-2017-11719](https://nvd.nist.gov/vuln/detail/CVE-2017-11719) CVSSv3=7.8 (nixos-unstable)
* [ ] [CVE-2016-6920](https://nvd.nist.gov/vuln/detail/CVE-2016-6920) CVSSv3=7.5 (nixos-unstable)
* [ ] [CVE-2018-15822](https://nvd.nist.gov/vuln/detail/CVE-2018-15822) CVSSv3=7.5 (nixos-unstable)
* [ ] [CVE-2017-9608](https://nvd.nist.gov/vuln/detail/CVE-2017-9608) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2017-15186](https://nvd.nist.gov/vuln/detail/CVE-2017-15186) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-6392](https://nvd.nist.gov/vuln/detail/CVE-2018-6392) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-6621](https://nvd.nist.gov/vuln/detail/CVE-2018-6621) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-6912](https://nvd.nist.gov/vuln/detail/CVE-2018-6912) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-7557](https://nvd.nist.gov/vuln/detail/CVE-2018-7557) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-7751](https://nvd.nist.gov/vuln/detail/CVE-2018-7751) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-10001](https://nvd.nist.gov/vuln/detail/CVE-2018-10001) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-14394](https://nvd.nist.gov/vuln/detail/CVE-2018-14394) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-14395](https://nvd.nist.gov/vuln/detail/CVE-2018-14395) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-1999012](https://nvd.nist.gov/vuln/detail/CVE-2018-1999012) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-1999013](https://nvd.nist.gov/vuln/detail/CVE-2018-1999013) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-1999014](https://nvd.nist.gov/vuln/detail/CVE-2018-1999014) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2018-1999015](https://nvd.nist.gov/vuln/detail/CVE-2018-1999015) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2016-6881](https://nvd.nist.gov/vuln/detail/CVE-2016-6881) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-7122](https://nvd.nist.gov/vuln/detail/CVE-2016-7122) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-7555](https://nvd.nist.gov/vuln/detail/CVE-2016-7555) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-7562](https://nvd.nist.gov/vuln/detail/CVE-2016-7562) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-7785](https://nvd.nist.gov/vuln/detail/CVE-2016-7785) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-7905](https://nvd.nist.gov/vuln/detail/CVE-2016-7905) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-8595](https://nvd.nist.gov/vuln/detail/CVE-2016-8595) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2016-9561](https://nvd.nist.gov/vuln/detail/CVE-2016-9561) CVSSv3=5.5 (nixos-unstable)

Scanned versions: nixos-unstable: 28fce082c8c. May contain false positives.

Cc @codyopel

### LINKED ISSUE #120371: Vulnerability roundup 101: ffmpeg-2.8.17: 1 advisory [8.8]
[search](https://search.nix.gsc.io/?q=ffmpeg&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=ffmpeg+in%3Apath&type=Code)

* [ ] [CVE-2021-30123](https://nvd.nist.gov/vuln/detail/CVE-2021-30123) CVSSv3=8.8 (nixos-20.09, nixos-unstable)

Scanned versions: nixos-20.09: c7e905b6a97; nixos-unstable: f5e8bdd07d1.

Cc @codyopel

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
It's outdated and insecure.
closes https://github.com/NixOS/nixpkgs/issues/93984 and closes https://github.com/NixOS/nixpkgs/issues/120371

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I did not test whether the packages actually still work, just whether they build and run.

Two packages remain:
* `stepmania` might compile with a newer Ffmpeg if we use a more recent revision (no release made though) or with the vendored Ffmpeg. cc @RubenAstudillo
Package needs a new maintainer.
* `pcsxr` should also work if updated to 1.9.95. See https://aur.archlinux.org/packages/pcsxr. I gave it a try but stopped after some complications. cc @rardiol

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24092,11 +24092,6 @@ in
 
   kile = libsForQt5.callPackage ../applications/editors/kile { };
 
-  kino = callPackage ../applications/video/kino {
```
Comment (ajs124) [inline pkgs/top-level/all-packages.nix:24098]: You seem to have forgotten to actually delete this folder?

Comment (dotlambda) [reply pkgs/top-level/all-packages.nix:24098]: indeed

**Comment by Profpatsch** [comment]
Does ultrastardx still work with ffmpeg3?

**Comment by dotlambda** [comment]
> Does ultrastardx still work with ffmpeg3?

I was able to build and run it. It played some sounds but I didn't test any functionality.
However, we are also trying to get rid of `ffmpeg_3`: https://github.com/NixOS/nixpkgs/issues/120705

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 2]
```diff
@@ -17,7 +17,7 @@
 # Darwin frameworks
 , Cocoa, darwinFrameworks ? [ Cocoa ]
 # Inherit generics
-, branch, sha256, version, patches ? [], ...
+, branch, sha256, version, patches ? [], knownVulnerabilities ? [ ], ...
```
Comment (lukegb) [inline pkgs/development/libraries/ffmpeg/generic.nix:20]: Just `[]` for consistency?

Comment (dotlambda) [reply pkgs/development/libraries/ffmpeg/generic.nix:20]: Sure,even though I see no value in it.

**Review by lukegb — APPROVED** [review]

**Comment by dotlambda** [comment]
backport: https://github.com/NixOS/nixpkgs/pull/121908

**Pull request merged**